src/components/SettingsModal.js

```diff
@@ -0,0 +1,159 @@
+    } catch (error) {
+      setErrorMessage(error.message);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+  return (
+    <Modal open={isModalOpen} onClose={() => setIsModalOpen(false)}>
+      <Box sx={settingsModalStyle}>
+        <Typography variant={componentProps.variant.h4} color={componentProps.color.primary} sx={{ mb: 4 }} component={componentProps.variant.h1}>
+          Definiçoes gerais
+        </Typography>
+        <Box sx={settingsClasses.form}>
+          <form noValidate onSubmit={handleSubmit(handleSubmitSettings)}>
+            <Controller
+              name={componentProps.name.minAmountForFreeDelivery}
+              control={control}
+              rules={{ required: 'Valor minimo em falta' }}
+              render={({ field }) => (
+                <TextField
+                  label='Valor minimo para entrega grátis'
+                  type={componentProps.type.text}
+                  variant={componentProps.variant.outlined}
+                  fullWidth
+                  sx={settingsClasses.formField}
+                  error={errors.minAmountForFreeDelivery ? true : false}
+                  autoComplete='true'
+                  {...field}
+                />
+              )}
+            />
+
+            <Controller
+              name={componentProps.name.deliveryFee}
+              control={control}
+              rules={{ required: 'Valor de taxa entrega em falta' }}
+              render={({ field }) => (
+                <TextField
+                  label='Valor de taxa entrega'
+                  type={componentProps.type.text}
+                  variant={componentProps.variant.outlined}
+                  fullWidth
+                  sx={settingsClasses.formField}
+                  error={errors.deliveryFee ? true : false}
+                  autoComplete='true'
+                  {...field}
+                />
+              )}
+            />
+
+            <Controller
+              name={componentProps.name.discount}
+              control={control}
+              rules={{ required: 'Desconto em falta' }}
+              render={({ field }) => (
+                <TextField
+                  label='Percentagem de desconto'
+                  type={componentProps.type.text}
+                  variant={componentProps.variant.outlined}
+                  fullWidth
+                  sx={settingsClasses.formField}
+                  error={errors.discount ? true : false}
+                  autoComplete='true'
+                  {...field}
+                />
+              )}
+            />
+
+            <Controller
+              control={control}
+              name={componentProps.name.haveGlobalDeliveryDiscount}
+              render={({ field }) => <FormControlLabel control={<Switch checked={haveGlobalDeliveryDiscount} {...field} />} label='Entrega gratuita geral' />}
+            />
+
+            <button type={componentProps.type.submit} ref={submitFormRef} hidden>
```

Comment:
Hidden button?

Reply:
Only because i am displaying another button instead and the form must have a button. but yeah i could move the other inside and remove that one.